pkg/virt-controller/watch/clone/clone_test.go

```diff
@@ -1002,6 +1002,39 @@ var _ = Describe("Clone", func() {
 
 		})
 
+		Context("HostName", func() {
+
+			const manuallySetHostName = "manually-set-hostname"
+
+			BeforeEach(func() {
+				sourceVM.Spec.Template.Spec.Hostname = vmClone.Name
+			})
+
+			expectHostName := func(hostname string) {
+				expectedVM := sourceVM.DeepCopy()
+				expectedVM.Spec.Template.Spec.Hostname = hostname
+				expectVMCreationFromPatches(expectedVM)
+			}
+
+			It("should set with the kubevirt generated hostname if hostname is not provided", func() {
+				addClone(vmClone)
+				expectHostName(vmClone.Name)
+				expectSnapshotContentGet(sourceVM)
+
+				controller.Execute()
+			})
```

Comment:
The `BeforeEach` instruction makes the test `should set with the kubevirt generated hostname if hostname is not provided` not correct.
In the `BeforeEach` you are providing an hostname to the target vmi.
Please, drop the `BeforeEach`.
Thanks